Ci is failing in mypy checks

Nighlty CI is failing as follows
```
Error: qiskit_nature/second_q/hamiltonians/lattices/lattice.py:25: error: Unused "type: ignore" comment  [unused-ignore]
Error: qiskit_nature/second_q/hamiltonians/lattices/hexagonal_lattice.py:17: error: Unused "type: ignore" comment  [unused-ignore]
Error: test/second_q/hamiltonians/lattices/test_triangular_lattice.py:17: error: Unused "type: ignore" comment  [unused-ignore]
Error: test/second_q/hamiltonians/lattices/test_square_lattice.py:17: error: Unused "type: ignore" comment  [unused-ignore]
Error: test/second_q/hamiltonians/lattices/test_line_lattice.py:17: error: Unused "type: ignore" comment  [unused-ignore]
Error: test/second_q/hamiltonians/lattices/test_lattice.py:20: error: Unused "type: ignore" comment  [unused-ignore]
Error: test/second_q/hamiltonians/lattices/test_kagome_lattice.py:18: error: Unused "type: ignore" comment  [unused-ignore]
Error: test/second_q/hamiltonians/lattices/test_hyper_cubic_lattice.py:17: error: Unused "type: ignore" comment  [unused-ignore]
Error: test/second_q/hamiltonians/lattices/test_hexagonal_lattice.py:17: error: Unused "type: ignore" comment  [unused-ignore]
Error: test/second_q/hamiltonians/test_ising_model.py:19: error: Unused "type: ignore" comment  [unused-ignore]
Error: test/second_q/hamiltonians/test_heisenberg_model.py:16: error: Unused "type: ignore" comment  [unused-ignore]
Error: test/second_q/hamiltonians/test_fermi_hubbard_model.py:18: error: Unused "type: ignore" comment  [unused-ignore]
```
which seems attributable to yesterdays (23 Jan) release of a new rustworkx version. The prior run that passed was on 0.13.2 and the new version is 0.14.0

> This is a new feature release of Rustworkx that adds many new features to the library. The highlights of this release are:
>
>        * Fully type annotated for support with [mypy](https://mypy-lang.org/) and other tooling

so presumably with the annotation the ignore is now in conflict.

--- a/qiskit_nature/second_q/hamiltonians/lattices/hexagonal_lattice.py
+++ b/qiskit_nature/second_q/hamiltonians/lattices/hexagonal_lattice.py
@@ -1,6 +1,6 @@
 # This code is part of a Qiskit project.
 #
-# (C) Copyright IBM 2021, 2023.
+# (C) Copyright IBM 2021, 2024.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory
@@ -14,7 +14,7 @@
 
 from __future__ import annotations
 
-from rustworkx import generators  # type: ignore[attr-defined]
+from rustworkx import generators
 
 from .lattice import Lattice
 

--- a/qiskit_nature/second_q/hamiltonians/lattices/lattice.py
+++ b/qiskit_nature/second_q/hamiltonians/lattices/lattice.py
@@ -1,6 +1,6 @@
 # This code is part of a Qiskit project.
 #
-# (C) Copyright IBM 2021, 2023.
+# (C) Copyright IBM 2021, 2024.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory
@@ -22,7 +22,7 @@ import numbers
 import numpy as np
 
 from rustworkx import NodeIndices, PyGraph, WeightedEdgeList
-from rustworkx import adjacency_matrix, networkx_converter  # type: ignore[attr-defined]
+from rustworkx import adjacency_matrix, networkx_converter
 from rustworkx.visualization import mpl_draw
 
 from qiskit.utils import optionals as _optionals

--- a/test/second_q/hamiltonians/lattices/test_hexagonal_lattice.py
+++ b/test/second_q/hamiltonians/lattices/test_hexagonal_lattice.py
@@ -1,6 +1,6 @@
 # This code is part of a Qiskit project.
 #
-# (C) Copyright IBM 2021, 2023.
+# (C) Copyright IBM 2021, 2024.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory
@@ -14,7 +14,7 @@
 from test import QiskitNatureTestCase
 import numpy as np
 from numpy.testing import assert_array_equal
-from rustworkx import PyGraph, is_isomorphic  # type: ignore[attr-defined]
+from rustworkx import PyGraph, is_isomorphic
 from qiskit_nature.second_q.hamiltonians.lattices import HexagonalLattice
 
 

--- a/test/second_q/hamiltonians/lattices/test_hyper_cubic_lattice.py
+++ b/test/second_q/hamiltonians/lattices/test_hyper_cubic_lattice.py
@@ -1,6 +1,6 @@
 # This code is part of a Qiskit project.
 #
-# (C) Copyright IBM 2021, 2023.
+# (C) Copyright IBM 2021, 2024.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory
@@ -14,7 +14,7 @@
 from test import QiskitNatureTestCase
 import numpy as np
 from numpy.testing import assert_array_equal
-from rustworkx import PyGraph, is_isomorphic  # type: ignore[attr-defined]
+from rustworkx import PyGraph, is_isomorphic
 from qiskit_nature.second_q.hamiltonians.lattices import (
     BoundaryCondition,
     HyperCubicLattice,

--- a/test/second_q/hamiltonians/lattices/test_kagome_lattice.py
+++ b/test/second_q/hamiltonians/lattices/test_kagome_lattice.py
@@ -1,6 +1,6 @@
 # This code is part of a Qiskit project.
 #
-# (C) Copyright IBM 2021, 2023.
+# (C) Copyright IBM 2021, 2024.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory
@@ -15,7 +15,7 @@ import unittest
 from test import QiskitNatureTestCase
 from numpy.testing import assert_array_equal
 import numpy as np
-from rustworkx import PyGraph, is_isomorphic  # type: ignore[attr-defined]
+from rustworkx import PyGraph, is_isomorphic
 from qiskit_nature.second_q.hamiltonians.lattices import (
     BoundaryCondition,
     KagomeLattice,

--- a/test/second_q/hamiltonians/lattices/test_lattice.py
+++ b/test/second_q/hamiltonians/lattices/test_lattice.py
@@ -1,6 +1,6 @@
 # This code is part of a Qiskit project.
 #
-# (C) Copyright IBM 2021, 2023.
+# (C) Copyright IBM 2021, 2024.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory
@@ -17,7 +17,7 @@ import unittest
 import numpy as np
 from numpy.testing import assert_array_equal
 
-from rustworkx import PyGraph, is_isomorphic  # type: ignore[attr-defined]
+from rustworkx import PyGraph, is_isomorphic
 
 from qiskit.utils import optionals as _optionals
 

--- a/test/second_q/hamiltonians/lattices/test_line_lattice.py
+++ b/test/second_q/hamiltonians/lattices/test_line_lattice.py
@@ -1,6 +1,6 @@
 # This code is part of a Qiskit project.
 #
-# (C) Copyright IBM 2021, 2023.
+# (C) Copyright IBM 2021, 2024.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory
@@ -14,7 +14,7 @@
 from test import QiskitNatureTestCase
 from numpy.testing import assert_array_equal
 import numpy as np
-from rustworkx import PyGraph, is_isomorphic  # type: ignore[attr-defined]
+from rustworkx import PyGraph, is_isomorphic
 from qiskit_nature.second_q.hamiltonians.lattices import (
     BoundaryCondition,
     LineLattice,

--- a/test/second_q/hamiltonians/lattices/test_square_lattice.py
+++ b/test/second_q/hamiltonians/lattices/test_square_lattice.py
@@ -1,6 +1,6 @@
 # This code is part of a Qiskit project.
 #
-# (C) Copyright IBM 2021, 2023.
+# (C) Copyright IBM 2021, 2024.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory
@@ -14,7 +14,7 @@
 from test import QiskitNatureTestCase
 from numpy.testing import assert_array_equal
 import numpy as np
-from rustworkx import PyGraph, is_isomorphic  # type: ignore[attr-defined]
+from rustworkx import PyGraph, is_isomorphic
 from qiskit_nature.second_q.hamiltonians.lattices import (
     BoundaryCondition,
     SquareLattice,

--- a/test/second_q/hamiltonians/lattices/test_triangular_lattice.py
+++ b/test/second_q/hamiltonians/lattices/test_triangular_lattice.py
@@ -1,6 +1,6 @@
 # This code is part of a Qiskit project.
 #
-# (C) Copyright IBM 2021, 2023.
+# (C) Copyright IBM 2021, 2024.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory
@@ -14,7 +14,7 @@
 from test import QiskitNatureTestCase
 from numpy.testing import assert_array_equal
 import numpy as np
-from rustworkx import PyGraph, is_isomorphic  # type: ignore[attr-defined]
+from rustworkx import PyGraph, is_isomorphic
 from qiskit_nature.second_q.hamiltonians.lattices import (
     BoundaryCondition,
     TriangularLattice,

--- a/test/second_q/hamiltonians/test_fermi_hubbard_model.py
+++ b/test/second_q/hamiltonians/test_fermi_hubbard_model.py
@@ -1,6 +1,6 @@
 # This code is part of a Qiskit project.
 #
-# (C) Copyright IBM 2021, 2023.
+# (C) Copyright IBM 2021, 2024.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory
@@ -15,7 +15,7 @@ from test import QiskitNatureTestCase
 
 import numpy as np
 from numpy.testing import assert_array_equal
-from rustworkx import PyGraph, is_isomorphic  # type: ignore[attr-defined]
+from rustworkx import PyGraph, is_isomorphic
 
 from qiskit_nature.second_q.hamiltonians import FermiHubbardModel
 from qiskit_nature.second_q.hamiltonians.lattices import Lattice

--- a/test/second_q/hamiltonians/test_heisenberg_model.py
+++ b/test/second_q/hamiltonians/test_heisenberg_model.py
@@ -1,6 +1,6 @@
 # This code is part of a Qiskit project.
 #
-# (C) Copyright IBM 2022, 2023.
+# (C) Copyright IBM 2022, 2024.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory
@@ -13,7 +13,7 @@
 """Test HeisenbergModel."""
 
 from test import QiskitNatureTestCase
-from rustworkx import PyGraph, is_isomorphic  # type: ignore[attr-defined]
+from rustworkx import PyGraph, is_isomorphic
 from qiskit_nature.second_q.hamiltonians.lattices import Lattice, LineLattice
 from qiskit_nature.second_q.hamiltonians import HeisenbergModel, IsingModel
 

--- a/test/second_q/hamiltonians/test_ising_model.py
+++ b/test/second_q/hamiltonians/test_ising_model.py
@@ -1,6 +1,6 @@
 # This code is part of a Qiskit project.
 #
-# (C) Copyright IBM 2021, 2023.
+# (C) Copyright IBM 2021, 2024.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory
@@ -16,7 +16,7 @@ from test import QiskitNatureTestCase
 
 import numpy as np
 from numpy.testing import assert_array_equal
-from rustworkx import PyGraph, is_isomorphic  # type: ignore[attr-defined]
+from rustworkx import PyGraph, is_isomorphic
 
 from qiskit_nature.second_q.hamiltonians.lattices import Lattice
 from qiskit_nature.second_q.hamiltonians import IsingModel

